src/job/job.ts

```diff
@@ -0,0 +1,36 @@
+import { JobSummary } from './summary'
+
+/**
+ * Asynchronous task hanlder.
+ *
+ * Status workflow:
+ * - pending
+ * - running
+ * - (end)
+ * -- succeed
+ * -- failed
+ * -- canceled
+ *  */
+export interface Job<P, R, C> {
+  /**
```

Comment:
don't we have inner jobs here?

Reply:
uhm, should inner jobs be in the "job summary" or whatever it is?